src/views/Nft/market/hooks/useNftsForAddress.tsx

```diff
@@ -26,13 +27,11 @@ const useNftsForAddress = (account: string, profile: Profile, isProfileFetching:
     return null
   }, [profileNftTokenId, profileNftCollectionAddress, hasProfileNft])
 
-  const { status, data, mutate } = useSWR([account, collections, isProfileFetching, 'userNfts'], async () => {
-    if (!isProfileFetching && !isEmpty(collections) && isAddress(account)) {
-      return getCompleteAccountNftData(account, collections, profileNftWithCollectionAddress)
-    }
-    return []
-  })
-
+  const { status, data, mutate } = useSWR(
+    !isProfileFetching && !isEmpty(collections) && isAddress(account) ? [account, 'userNfts'] : null,
+    async () => getCompleteAccountNftData(account, collections, profileNftWithCollectionAddress),
+    { use: [laggyMiddleware] },
```

Comment:
I think the reason we need laggy here is because `isProfileFetching` toggle key changes. Can't think of any better way yet but sure it's a problem